### PR TITLE
Candlestick: fix auto field mapping & includeAllFields

### PR DESCRIPTION
**Pull request by leeoniya**
followup to https://github.com/grafana/grafana/pull/42011, which i didn't test sufficiently :(

- fixes a crash when `field.state` was undefined
- does not purge high/low fields in volume-only mode if they're also auto-mapped to open/close
- fixes a stupid `includeAllFields` bug with fields not being included when timeField is at index 0 (refactoring oversight)
- removes all transforms from Candlestick gdev dashboard that were needed previously

additionally (i dont want to make an extra PR):

the `fillBelowTo` bugfix in https://github.com/grafana/grafana/pull/41998 uncovered the fact that we had insufficient config for bands in a couple gdev dashboards and were relying on the bug for them to look the way they did. the two dashboards are updated to explicitly set `fillOpacity` to 0 for the field that goes through the center of the bands.

PS: @ryantxu is out for the next week, so he's unlikely to review this soon.

cc @wardbekker

### CODE REVIEW

File: public/app/plugins/panel/candlestick/fields.ts [hunk 1]
```diff
@@ -184,11 +184,15 @@ export function prepareCandlestickFields(
...
+      }
       data.high = undefined;
     }
     if (data.low) {
-      used.delete(data.low);
+      if (data.low !== data.open) {
+        used.delete(data.low);
+      }
       data.low = undefined;
     }
```
Comment (torkelo) [inline public/app/plugins/panel/candlestick/fields.ts:197]: So this function has unit tests, the nice thing with that is now that we identified a bug/crash/new case we can just add a new scenario that triggers the error, then add the code fixes it :) 
https://github.com/grafana/grafana/blob/f858b483c634cbf54f9683775b2edef377aa08fc/public/app/plugins/panel/candlestick/fields.test.ts#L57

Comment (leeoniya) [reply public/app/plugins/panel/candlestick/fields.ts:197]: added some tests

**Review by torkelo — APPROVED** [review]
Looks great

**Comment by leeoniya** [comment]
thanks for the review!

**Pull request merged**